### PR TITLE
[ROCm] Update ROCm CI to python3.6

### PR DESCRIPTION
**Pull request by iotamudelta**
Rehash of #22322 .

Given that python 2.7 will be EOL'd on Jan 1, 2020 and we have models depending on python3.5+, we'd like to update the ROCm CI across the board to python3.6.

This PR adds the skip tests and some semantic changes for PyTorch.

Added pattern match skip for anything but the ROCm CI compared to #223222 for the python find step in the PyTorch build.

### CODE REVIEW

**Comment by iotamudelta** [comment]
ping @bddppq

File: .jenkins/pytorch/common.sh [hunk 1]
```diff
@@ -23,6 +23,12 @@ SCRIPT_DIR="$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )"
 # Required environment variables:
 #   $BUILD_ENVIRONMENT (should be set by your Docker image)
 
+# Figure out which Python to use
+PYTHON="$(which python)"
```
Comment (bddppq) [inline .jenkins/pytorch/common.sh]: This line is not needed

File: .jenkins/pytorch/common.sh [hunk 2]
```diff
@@ -23,6 +23,12 @@ SCRIPT_DIR="$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )"
...
+if [[ "${BUILD_ENVIRONMENT}" =~ py((2|3)\.?[0-9]?\.?[0-9]?) ]] && [[ "${BUILD_ENVIRONMENT}" == *rocm* ]]; then
```
Comment (bddppq) [inline .jenkins/pytorch/common.sh]: nit: put the rocm check first to make it more obvious

File: .jenkins/pytorch/test.sh [hunk 3]
```diff
@@ -11,6 +11,14 @@ source "$(dirname "${BASH_SOURCE[0]}")/common.sh"
 
 echo "Testing pytorch"
 
+# this needs to happen first
+if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
+  # non-interactive bashs do not expand aliases by default
+  shopt -s expand_aliases
+  export PYTORCH_TEST_WITH_ROCM=1
+  alias python="$PYTHON"
```
Comment (bddppq) [inline .jenkins/pytorch/test.sh]: better to move the alias to common.sh

File: .jenkins/pytorch/build.sh [hunk 4]
```diff
@@ -104,10 +104,10 @@ if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
     export PATH="$CACHE_WRAPPER_DIR:$PATH"
   fi
 
-  python tools/amd_build/build_amd.py
+  $PYTHON tools/amd_build/build_amd.py
```
Comment (bddppq) [inline .jenkins/pytorch/build.sh]: This is not needed anymore since the alias is now in common.sh

**Review by bddppq — APPROVED** [review]
LGTM

**Comment by bddppq** [comment]
@iotamudelta the windows CI failure seems to be consistent, could you merge your PR with master?
btw. if change your repo's setting to grant write access to upstream admin, we can use pytorchbot to automatically doing the rebase/merge with master thing https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request closed without merging**